### PR TITLE
Remove Haystack code, flag references, index file, and tests from TDP

### PR DESCRIPTION
**Pull request by higs4281**
Elasticsearch7 is live, so we can remove the Haystack code, files, tests and Flag from TDP.

## How to test this PR
- Run unittest-current and check 100% coverage
```
tox -e unittest-current teachers_digital_platform
coverage report -m | egrep "Stmts|teach"
```

Check out the local activities search page (Docker setup) and make sure search and faceting work as expected.
<http://localhost:8000/consumer-tools/educator-tools/youth-financial-education/teach/activities/>

### CODE REVIEW

**Review by cwdavies — APPROVED** [review]

File: cfgov/teachers_digital_platform/models/activity_index_page.py [hunk 1]
```diff
@@ -210,186 +208,14 @@ def dsl_search(self, request, *args, **kwargs):
...
-            'current_page': current_page,
-            'paginator': paginator,
-            'show_filters': bool(facet_queries),
-        }
-        return context_update
-
     def get_context(self, request, *args, **kwargs):
         if not self.activity_setups:
             self.activity_setups = get_activity_setup()
-        if flag_enabled('ELASTICSEARCH_DSL_TDP'):
```
Comment (Scotchester) [inline cfgov/teachers_digital_platform/models/activity_index_page.py:302]: Don't forget to remove the flag definition in `settings/base.py`.

Comment (higs4281) [reply cfgov/teachers_digital_platform/models/activity_index_page.py:302]: I thought we'd leave flag declarations alone until all apps are clear. Too cautious?

**Review by alexm118 — APPROVED** [review]
LGTM

File: cfgov/teachers_digital_platform/models/activity_index_page.py (hunk 1 above)
Comment (Scotchester) [reply cfgov/teachers_digital_platform/models/activity_index_page.py:302]: I guess I'd say if you're removing the code that uses the flag, might as well remove the declaration.

Comment (higs4281) [reply cfgov/teachers_digital_platform/models/activity_index_page.py:302]: Sold.

**Pull request merged**